File: 4st/Eunseong/콜라츠 수열 만들기.java
class Solution {
    public int[] solution(int n) {
        int cnt = 0;
        int sum = n;
        while (sum != 1) {
            if (sum % 2 == 0) {
                sum /= 2;
            } else {
                sum *= 3;
                sum++;
            }
            cnt++;
        }
        int[] answer = new int[++cnt];
        sum = n;
        for (int i = 0; i < answer.length; i++) {
            answer[i] = sum;
            if (sum % 2 == 0) {
                sum /= 2;
            } else {
                sum *= 3;
                sum++;
            }
        }
        return answer;
    }
}

File: 3st/Juram/programmer/콜라츠 수열 만들기.js
function solution(n) {
    var answer = [];
    answer.push(n);
    while(true){
        if(n===1){
            break;
        } 
        if(n%2==0){
            n = n/2;
            answer.push(n);
        }else{
            n = 3*n+1;
            answer.push(n);
        }
    }
    return answer;
}